homeassistant/components/vizio/media_player.py

```diff
@@ -61,6 +61,11 @@
             config_entry, options={CONF_VOLUME_STEP: volume_step}
         )
 
+    if not config_entry.data.get(CONF_VOLUME_STEP):
+        new_data = config_entry.data.copy()
+        new_data.update({CONF_VOLUME_STEP: volume_step})
+        hass.config_entries.async_update_entry(config_entry, data=new_data)
```

Comment:
Can we consolidate the call to update the entry with the logic above so that we only do max one call?

```py
params = {}

if not config_entry.options::
    ...
    params["options"] = options
if not config_entry.data.get(CONF_VOLUME_STEP):
    ...
    params["data"] = new_data

if params:
    hass.config_entries.async_update_entry(config_entry, **params)
```

Reply:
good idea